module/zfs/dmu_tx.c

```diff
@@ -776,14 +776,16 @@ static const hrtime_t zfs_delay_max_ns = 100 * MICROSEC; /* 100 milliseconds */
  * of zfs_delay_scale to increase the steepness of the curve.
  */
 static void
-dmu_tx_delay(dmu_tx_t *tx, uint64_t dirty)
+dmu_tx_delay(dmu_tx_t *tx, uint64_t dirty, hrtime_t last_smooth)
 {
 	dsl_pool_t *dp = tx->tx_pool;
 	uint64_t delay_min_bytes =
 	    zfs_dirty_data_max * zfs_delay_min_dirty_percent / 100;
-	hrtime_t wakeup, min_tx_time, now;
+	hrtime_t wakeup, min_tx_time, now, smoothing_time, delay_time;
 
-	if (dirty <= delay_min_bytes)
+	now = gethrtime();
+
+	if (dirty <= delay_min_bytes && last_smooth <= now)
```

Comment:
We can avoid this early exit if we can take the "performance hit" of the two zero initializations in line 799 and 800, and the MIN(MAX(...)) code in line 811.

Reply:
Doesn't it make code more readable?